hilti/runtime/include/util.h

```diff
@@ -649,4 +650,97 @@ constexpr std::size_t hashCombine(std::size_t hash1, std::size_t hash2, Hashes..
         return result;
 }
 
+template<typename Data, typename Error>
+class ControlBlockRef;
+
+// Helper class for loosely tracking liveliness of some memory. Code using this
+// would instantiate an instance of this pointing to some data. It can then
+// pass out references to that control block by calling `asRef`. One can get
+// values from these references as long as the original control block is live.
+template<typename Data, typename Error>
+class ControlBlock {
```

Comment:
How about moving the new classes into their own `rt::util::control` namespace (`control::Block`, `control::Reference`)? That's closer to how we structure things elsewhere, and a bit more clear too I'd say.